### PR TITLE
Import all resources by default

### LINKED ISSUE #4: Import all resources by default
As a platform engineer I want to easily use function-aws-importer without the need to tell it which resources to import because in most cases I just want to import all of them.

**Acceptance Criteria**
- imports all resources by default
- makes `resourceName` from input optional

### PR DESCRIPTION
**Pull request by LCaparelli**
Big refac, suggest looking one commit at a time, each tells a more digestible story.

Fixes #4

### CODE REVIEW

File: fn.go [hunk 1]
```diff
@@ -58,148 +56,115 @@ func (f *Function) RunFunction(ctx context.Context, req *fnv1beta1.RunFunctionRe
...
-
-	desiredComposed, ok := desiredMRs[in.ResourceName]
-	if !ok {
-		f.log.Info("Failed to get desired Composed Resource from req.",
-			"error", err,
-			"resourceName", in.ResourceName,
-		)
-		response.Fatal(rsp, fmt.Errorf("cannot get desired MR %q from req", in.ResourceName))
-		return rsp, nil
+	// TODO(lcarelli): handle early return when desired is empty. If you spot this on review, please mention it.
```
Comment (go-faustino) [inline fn.go:68]: ☝🏻

File: fn.go [hunk 2]
```diff
@@ -58,148 +56,115 @@ func (f *Function) RunFunction(ctx context.Context, req *fnv1beta1.RunFunctionRe
...
-		f.log.Info("Ambiguous tag filters.", // TODO(lcaparelli): better word than ambiguous, maybe better message overall
-			"error", errors.New("found more than one resource matching tag filters"),
-			"tagFilters", tagFilters,
-			"matchingResources", extractARNs(tagMappings),
-		)
-		response.Fatal(rsp, fmt.Errorf("found more than one resource matching tag filters: %v", extractARNs(tagMappings)))
-		return rsp, nil
-	}
+		if len(tagMappings) > 1 {
+			f.log.Info("Ambiguous tag filters.", // TODO(lcaparelli): better word than ambiguous, maybe better message overall
```
Comment (go-faustino) [inline fn.go:113]: Maybe nonunique?

**Review by go-faustino — APPROVED** [review]
LGTM

File: fn.go (hunk 2 above)
Comment (LCaparelli) [reply fn.go:113]: The result is non-unique, but not the tag filters themselves... :thinking: 

Also thought about "Unresolvable tag filters", but they are resolvable. In fact more than once, hence the problem lol

Maybe "Cannot decide which resource to import."?

Comment (LCaparelli) [reply fn.go:117]: > Maybe "Cannot decide which resource to import."?

Running with that one, let me know if you can come up with a better one and we can adjust it :grin:

**Review by caiofralmeida — APPROVED** [review]

**Pull request merged**